src/app/donation-start/donation-start.component.ts

```diff
@@ -156,6 +156,8 @@ export class DonationStartComponent implements AfterContentChecked, AfterContent
     this.campaign = this.route.snapshot.data.campaign;
     this.setCampaignBasedVars();
 
+    const emailRegex : RegExp = /^(?=.{1,254}$)(?=[a-zA-Z0-9+=_.-]{1,64}@)[a-zA-Z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-zA-Z0-9!#$%&'*+/=?^_`{|}~-]+)*@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(\.[a-zA-Z]{2,})+$/;
```

Comment:
@NoelLH one of the disadvantages of keeping the original `(?=.{1,64}@)` is that it will mean the email _can_ have special characters in the username section. For example, `a&^^.!!!!$%li@thebiggive.org.uk` would be valid. If you're okay with this then sure I'll proceed to change it back to `(?=.{1,64}@)`.

Apologies I didn't get what you mean with your last sentence. Do you mean to clarify on Jira that the `\.` is no longer an issue due to using the `/^.../` syntax?

Reply:
@alihejazi I think that 'weird' characters in the username part are fine given they are allowed by the specs and most people won't use this anyway. If we are keeping support for a username part at all, which I think we might as well, then I think it makes sense to keep the pattern simple and the constraints as similar as possible to other implementations rather than inventing our own additional ones.

To your second point, I just meant that I was confused that your pattern after _"It’s now been changed to this:"_ does not match the PR – in terms of the dot escape (which is part of the pattern and not just a string escape) but also in other larger ways. This makes it quite confusing to tie the reasoning for the changes to the actual code being proposed and therefore to review it effectively.

Reply:
@NoelLH Ok, I will change the username part back to the original `(?=.{1,64}@)`

Regarding your second point, I just realised the reason why my comment on Jira doesn't match this PR was because Jira rendered out some characters that are used for styling. For example the regex has two pairs of _, which have been rendered out as they're used for italics, and had at least two occurrences of ` too. I've just changed the format to make the regexp show in code blocks, not embedded code within the comments.